### PR TITLE
Jarv/wheel

### PR DESCRIPTION
**Pull request by jarv**
Add wheel support for python libraries
- Introduces a new standalone role that creates wheel archives from requirements files and uploads them to s3
- edxapp now downloads a wheel archive and installs prior to installation from requirements
- cleans up the base package installs

### CODE REVIEW

File: playbooks/roles/edxapp/defaults/main.yml [hunk 1]
```diff
@@ -226,7 +226,107 @@ lms_preview_auth_config:
...
+  - python_openid
+  - python_termstyle
+  - pytz
+  - pytz
+  - PyYAML
+  - rednose
+  - requests
+  - scipy
+  - selenium
+  - selenium
```
Comment (e0d) [inline playbooks/roles/edxapp/defaults/main.yml]: dupe

Comment (jarv) [inline playbooks/roles/edxapp/defaults/main.yml]: fixed right as you commented :)

File: playbooks/roles/edxapp/defaults/main.yml [hunk 2]
```diff
@@ -291,75 +387,29 @@ sandbox_post_requirements:  "{{ edx_platform_code_dir }}/requirements/edx-sandbo
...
+  # apparmor
+  - apparmor-utils
+  # misc
+  - aspell
+  - curl
+  - ipython
   - npm
   - ntp
-  - openjdk-7-jdk
-  - openjdk-7-jre
```
Comment (e0d) [inline playbooks/roles/edxapp/defaults/main.yml:338]: Does this need to be here?  Isn't it going to pull a bunch of other stuff into system python?

aspell?

File: playbooks/roles/edxapp/tasks/deploy.yml [hunk 3]
```diff
@@ -89,7 +89,27 @@
...
+
+- name: Untar the wheel archive
+  shell: >
+    tar -xzf {{edxapp_wheel_archive}}.tar.gz 
+      chdir=/var/tmp 
+      creates=/var/tmp/{{edxapp_wheel_archive}}
+    
+- name: Install from wheel files
+  shell: >
+    {{venv_dir}}/bin/pip install --use-wheel --no-index --find-links=/var/tmp/{{edxapp_wheel_archive}} {{item}}
```
Comment (feanil) [inline playbooks/roles/edxapp/tasks/deploy.yml:121]: Why do this instead of setting the wheel parameters for the existing plays?

Comment (jarv) [inline playbooks/roles/edxapp/tasks/deploy.yml:121]: The wheel package list is not created dynamically

File: playbooks/roles/edxapp/defaults/main.yml (hunk 2 above)
Comment (jarv) [inline playbooks/roles/edxapp/defaults/main.yml:338]: removed aspell

**Comment by jbau** [comment]
very interesting, @jarv  Is there anything preventing us (or other instances) from sharing a common wheel file?

**Comment by jarv** [comment]
As long as you are using the Ubuntu precise base image you can.  The wheel files need to be compiled on the same architecture/OS which means we should probably organize them that way in S3.

**Pull request closed without merging**